dll/win32/shell32/dialogs/dialogs.cpp

```diff
@@ -1070,78 +1071,103 @@ int WINAPI RestartDialogEx(HWND hWndOwner, LPCWSTR lpwstrReason, DWORD uFlags, D
     return 0;
 }
 
-/* Functions used for log off dialog box */
+/* Functions and macros used for log off dialog box */
+#define IS_PRODUCT_VERSION_WORKSTATION  0x300
 
-BOOL DrawIconOnOwnerDrawnButtons(DRAWITEMSTRUCT* pdis)
+#define CX_BITMAP                       33
+#define CY_BITMAP                       33
+
+#define BUTTON_SWITCH_USER              0
+#define BUTTON_SWITCH_USER_PRESSED      (CY_BITMAP + BUTTON_SWITCH_USER)
+#define BUTTON_SWITCH_USER_FOCUSED      (CY_BITMAP + BUTTON_SWITCH_USER_PRESSED)
+#define BUTTON_LOG_OFF                  (CY_BITMAP + BUTTON_SWITCH_USER_FOCUSED)
+#define BUTTON_LOG_OFF_PRESSED          (CY_BITMAP + BUTTON_LOG_OFF)
+#define BUTTON_LOG_OFF_FOCUSED          (CY_BITMAP + BUTTON_LOG_OFF_PRESSED)
+
+BOOL
+DrawIconOnOwnerDrawnButtons(
+    DRAWITEMSTRUCT* pdis,
+    HBITMAP hBitmap)
 {
-    HICON hCurrIcon;
+    BOOL bRet = FALSE;
+    HDC hdcMem = NULL;
+    HBITMAP hbmOld = NULL;
+    int y = 0;
     RECT rect;
-    
+
+    hdcMem = CreateCompatibleDC(pdis->hDC);
+    hbmOld = (HBITMAP)SelectObject(hdcMem, hBitmap);
```

Comment:
What's the reasoning behind the cast in this function?

Reply:
@Fraizeraust I have to cast that from HGDIOBJ to HBITMAP otherwise I will get the compiler error.
```
../dll/win32/shell32/dialogs/dialogs.cpp: In function 'BOOL DrawIconOnOwnerDrawn
Buttons(DRAWITEMSTRUCT*, HBITMAP)':
../dll/win32/shell32/dialogs/dialogs.cpp:1099:26: error: invalid conversion from
 'HGDIOBJ' {aka 'void*'} to 'HBITMAP' {aka 'HBITMAP__*'} [-fpermissive]
     hbmOld = SelectObject(hdcMem, hBitmap);
              ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

Reply:
This code is C++. Type cast is needed.